src/Microsoft.ServiceFabric.Diagnostics/Metrics/Implementation/IFabricMeterProvider.cs

```diff
@@ -0,0 +1,27 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+#if NET
+using System.Runtime.InteropServices.Marshalling;
+#else
+using GeneratedComInterfaceAttribute = System.Runtime.InteropServices.ComImportAttribute;
+#endif
+
+namespace Microsoft.ServiceFabric.Diagnostics.Metrics.Implementation
+{
+    [GeneratedComInterface]
+    [Guid("15AD37D2-F641-4188-824B-0D68CB4F6C17")]
+    [InterfaceType(ComInterfaceType.InterfaceIsIUnknown)]
+    partial interface IFabricMeterProvider
+    {
+        [MethodImpl(MethodImplOptions.InternalCall, MethodCodeType = MethodCodeType.Runtime)]
+        [return: MarshalAs(UnmanagedType.Interface)]
```

Comment:
Could you remove `[MarshalAs(UnmanagedType.Interface)] and `[MethodImpl]` attributes? See my comments in other interop files for details.